URBNSwiftyConvenience/Classes/Border.swift

```diff
@@ -43,24 +59,23 @@ public final class Border: UIView {
     /// Creates a border with a color, width, and inset
     ///
     /// - Parameters:
-    ///   - color: color for the border
-    ///   - pixelWidth: the width of the border in pixels (not points)
-    ///   - insets: insets for the border
-    public init(color: UIColor, pixelWidth: CGFloat = 1, insets: UIEdgeInsets = .zero) {
-        self.pixelWidth = pixelWidth
-        self.insets = insets
+    ///   - borderStyle: viewModel containing the following
+    ///     - color: color for the border
+    ///     - pixelWidth: the width of the border in pixels (not points)
+    ///     - insets: insets for the border
+    init?(borderStyle: BorderStyle?) {
+        guard let borderStyle = borderStyle else {
+            return nil
+        }
+        
+        self.borderStyle = borderStyle
+        self.pixelWidth = borderStyle.pixelWidth
+        self.insets = borderStyle.insets
```

Comment:
Why store `pixelWidth`, and `insets` if they’realready stored within borderStyle?

Reply:
pixelWidth & insets do some additional work in there `didSet` methods so I left them as separate private variables.

Reply:
That shouldn’t matter. You can do `borderStyle.pixelWidth.pixelsToPoints`, and `borderStyle.insets.top`. It can add confusion about which pixelWidth & insets are “correct”